### ISSUE TITLE
Regression in argument passthru added in 2.0.0

### ISSUE THREAD

**Issue by tofi86** [tofi86/universalJavaApplicationStub#39]
@Albertus82 wrote in https://github.com/tofi86/universalJavaApplicationStub/pull/31#issuecomment-261797631

> Hi @tofi86, I upgraded to v2.0.0 but my application won't start anymore :( If I manually open the app folder and start universalJavaApplicationStub, then the app starts along with the terminal window. OS X El Capitan, italian language.

**Comment by tofi86** [comment]
Since running the Stub from INSIDE the app package is not the normal behaviour, can you please try to run the stub from the folder where the app is located?

```
$ ./your.app/Contents/MacOS/universalJavaApplicationStub
```

Is there any output in the terminal leading to error messages?

Can you post the essentials (anonymized?) of your `Info.plist` file for further investigation?

Thanks!

**Comment by Albertus82** [comment]
@tofi86 
If I issue the command from inside the desktop, where the application is located, it starts fine:
```
./Codec.app/Contents/MacOS/universalJavaApplicationStub
```
but if I try to launch it from another folder:

```
Albertos-Mac:~ alberto$ Desktop/Codec.app/Contents/MacOS/universalJavaApplicationStub 
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fff89a1548d, pid=1028, tid=0x0000000000000a0b
#
# JRE version: Java(TM) SE Runtime Environment (8.0_111-b14) (build 1.8.0_111-b14)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.111-b14 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libsystem_pthread.dylib+0x148d]  pthread_mutex_lock+0x0
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/alberto/Desktop/hs_err_pid1028.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Abort trap: 6
```

Stack trace:
```
Stack: [0x00007fff5a443000,0x00007fff5ac43000],  sp=0x00007fff5ac3f5b8,  free space=8177k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libsystem_pthread.dylib+0x148d]  pthread_mutex_lock+0x0
C  [CoreFoundation+0x2f559]  CFBundleGetIdentifier+0x9
C  [HIToolbox+0x9b5e]  _ZL24RegisterAsDockClientPrivv+0x1e
C  [HIToolbox+0x9b1c]  _RegisterSharedDockClientWithRunLoop+0x26
C  [AppKit+0x7e4e]  _NSDoOneTimeDockRegistration+0x43
C  [AppKit+0x7dc9]  -[NSApplication _registerWithDock]+0x45
C  [AppKit+0x5fd9]  -[NSApplication init]+0x86f
C  [AppKit+0x5587]  +[NSApplication sharedApplication]+0x8e
C  [libswt-pi-cocoa-4430.jnilib+0xef6e]  Java_org_eclipse_swt_internal_cocoa_OS_objc_1msgSend__JJ+0x2e
j  org.eclipse.swt.internal.cocoa.OS.objc_msgSend(JJ)J+0
j  org.eclipse.swt.internal.cocoa.NSApplication.sharedApplication()Lorg/eclipse/swt/internal/cocoa/NSApplication;+6
j  org.eclipse.swt.widgets.Display.createDisplay(Lorg/eclipse/swt/graphics/DeviceData;)V+164
j  org.eclipse.swt.widgets.Display.create(Lorg/eclipse/swt/graphics/DeviceData;)V+18
j  org.eclipse.swt.graphics.Device.<init>(Lorg/eclipse/swt/graphics/DeviceData;)V+165
j  org.eclipse.swt.widgets.Display.<init>(Lorg/eclipse/swt/graphics/DeviceData;)V+2
j  org.eclipse.swt.widgets.Display.<init>()V+2
j  org.eclipse.swt.widgets.Display.getDefault()Lorg/eclipse/swt/widgets/Display;+16
j  it.albertus.codec.Codec.main([Ljava/lang/String;)V+40
v  ~StubRoutines::call_stub
V  [libjvm.dylib+0x2ecf8a]
V  [libjvm.dylib+0x323d82]
V  [libjvm.dylib+0x31caf9]
C  [java+0x3931]  JavaMain+0x9c4
C  [java+0x5590]  -[JavaLaunchHelper launchJava:]+0x2a
C  [Foundation+0x7754e]  __NSThreadPerformPerform+0x117
C  [CoreFoundation+0xaa621]  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__+0x11
C  [CoreFoundation+0x89e1c]  __CFRunLoopDoSources0+0x22c
C  [CoreFoundation+0x8933f]  __CFRunLoopRun+0x39f
C  [CoreFoundation+0x88d38]  CFRunLoopRunSpecific+0x128
C  [java+0x6465]  CreateExecutionEnvironment+0x367
C  [java+0x218c]  JLI_Launch+0x7a0
C  [java+0x84c2]  main+0x65
C  [java+0x19e4]  start+0x34
C  0x000000000000000a

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.eclipse.swt.internal.cocoa.OS.objc_msgSend(JJ)J+0
j  org.eclipse.swt.internal.cocoa.NSApplication.sharedApplication()Lorg/eclipse/swt/internal/cocoa/NSApplication;+6
j  org.eclipse.swt.widgets.Display.createDisplay(Lorg/eclipse/swt/graphics/DeviceData;)V+164
j  org.eclipse.swt.widgets.Display.create(Lorg/eclipse/swt/graphics/DeviceData;)V+18
j  org.eclipse.swt.graphics.Device.<init>(Lorg/eclipse/swt/graphics/DeviceData;)V+165
j  org.eclipse.swt.widgets.Display.<init>(Lorg/eclipse/swt/graphics/DeviceData;)V+2
j  org.eclipse.swt.widgets.Display.<init>()V+2
j  org.eclipse.swt.widgets.Display.getDefault()Lorg/eclipse/swt/widgets/Display;+16
j  it.albertus.codec.Codec.main([Ljava/lang/String;)V+40
v  ~StubRoutines::call_stub
```
Thank you!

**Comment by tofi86** [comment]
I had no problem getting your released version 1.2.0 to run with version 2.0.0 of the stub file.
All I did was copying the app to "Codec 2.app" and changing the content of the stub file...

<img width="838" alt="bildschirmfoto 2016-11-21 um 20 11 46" src="https://cloud.githubusercontent.com/assets/365118/20496823/fdb990bc-b026-11e6-9e52-2b2aaad55133.png">

What Java version are you running?

**Comment by Albertus82** [comment]
@tofi86 JRE 1.8.0_111. You tried to double click the app icon? I see the icon in the dock for a few seconds, then it disappears. Try also to start the app from the home directory instead of Downloads. Thank you.

**Comment by tofi86** [comment]
sorry, my fault... :-( indeed I cannot run the app by double clicking... that's weird... 
by the way, it works from the commandline with JDK 112 as well...

will digg deeper...

**Comment by tofi86** [comment]
Okay, finally I found two issues:

1. A regression in PR #31 by @dbankieris when passing external arguments to the app which I didn't notice during testing
   * If no parameter is passed to the app (`$@` should to be empty) apple is passing a default parameter (e.g. `-psn_0_82759` in `$@`). No idea why!
2. If an external parameter is beeing passed to your "Codec" app, it won't start the GUI but only on the console which results in a closing app symbol...

So I'm going to fix the apple default parameter with version 2.0.1 and you should be able to use the stub in your app...

**Comment by tofi86** [comment]
More info:
* http://stackoverflow.com/questions/10242115/os-x-strange-psn-command-line-parameter-when-launched-from-finder
* http://mirror.informatimago.com/next/developer.apple.com/documentation/Carbon/Reference/Process_Manager/prmref_main/data_type_5.html#//apple_ref/doc/uid/TP30000208/C001951

**Comment by tofi86** [comment]
Okay, this should be fixed now on the develop branch. @Albertus82 can you please check and confirm it works for you?
Thanks!

**Comment by Albertus82** [comment]
@tofi86 Fixed, thank you a lot! Waiting for the release.

**Comment by tofi86** [comment]
Great! Will wait a couple more days for other regressions till Friday...

**Comment by tofi86** [comment]
@Albertus82 just released version 2.0.1. Sorry for the delay.

**Comment by tofi86** [comment]
Note to myself: This does not seem to happen on the latest OS X versions anymore...
